distil/primitives/prediction_expansion.py

```diff
@@ -0,0 +1,104 @@
+import os
+import logging
+import typing
+
+from d3m import container, utils as d3m_utils
+from d3m.metadata import base as metadata_base, hyperparams
+from d3m.primitive_interfaces import base, transformer
+from d3m.contrib.primitives import compute_scores
+
+import common_primitives
+from common_primitives import construct_predictions
+
+__all__ = ('PredictionExpansionPrimitive',)
+logger = logging.getLogger(__name__)
+
+Inputs = container.DataFrame
+Outputs = container.DataFrame
+
+class PredictionExpansionPrimitive(construct_predictions.ConstructPredictionsPrimitive):
+    """
+    A primitive which takes as input a DataFrame and outputs a DataFrame in Lincoln Labs predictions
+    format: first column is a d3mIndex column (and other primary index columns, e.g., for object detection
+    problem), and then predicted targets, each in its column, followed by optional confidence column(s).
+
+    If the reference dataset has a multi index, the predictions will be expanded to match the reference
+    dataset shape.
+
+    It supports both input columns annotated with semantic types (``https://metadata.datadrivendiscovery.org/types/PrimaryKey``,
+    ``https://metadata.datadrivendiscovery.org/types/PrimaryMultiKey``, ``https://metadata.datadrivendiscovery.org/types/PredictedTarget``,
+    ``https://metadata.datadrivendiscovery.org/types/Confidence``), or trying to reconstruct metadata.
+    This is why the primitive takes also additional input of a reference DataFrame which should
+    have metadata to help reconstruct missing metadata. If metadata is missing, the primitive
+    assumes that all ``inputs`` columns are predicted targets, without confidence column(s).
+    """
+
+    metadata = metadata_base.PrimitiveMetadata(
+        {
+            'id': '36bedff2-1c99-47f3-92a8-a581f35b1924',
+            'version': '0.1.0',
+            'name': "Expand group predictions",
+            'python_path': 'd3m.primitives.data_transformation.prediction_expansion.DistilPredictionExpansion',
+            'source': {
+                'name': 'Distil',
+                'contact': 'mailto:cbethune@uncharted.software',
+                'uris': [
+                    'https://github.com/uncharted-distil/distil-primitives/distil/primitives/prediction_expansion.py',
+                    'https://github.com/uncharted-distil/distil-primitives',
+                ],
+            },
+            'installation': [{
+               'type': metadata_base.PrimitiveInstallationType.PIP,
+               'package_uri': 'git+https://gitlab.com/datadrivendiscovery/common-primitives.git@{git_commit}#egg=common_primitives'.format(
```

Comment:
Copy-paste issue - should point to `distil-primitives`

Reply:
latest commit has updated the reference